### PR TITLE
Added support for important mark (!) analysis.

### PR DESCRIPTION
**Pull request by dgquintas**
For example, an input of
```
GRPC_LAT_PROF 1316908107613.015869 0x7fd35f7fe700 { 205 (nil)
src/core/iomgr/tcp_posix.c 1
GRPC_LAT_PROF 1316908107614.015869 0x7fd35f7fe700 { 205 (nil)
src/core/iomgr/tcp_posix.c 2
GRPC_LAT_PROF 1316908107615.015869 0x7fd35f7fe700 ! 999999 (nil)
src/core/iomgr/tcp_posix.c 3
GRPC_LAT_PROF 1316908107616.015869 0x7fd35f7fe700 } 205 (nil)
src/core/iomgr/tcp_posix.c 4
GRPC_LAT_PROF 1316908107617.015869 0x7fd35f7fe700 ! 999999 (nil)
src/core/iomgr/tcp_posix.c 5
GRPC_LAT_PROF 1316908107618.015869 0x7fd35f7fe700 } 205 (nil)
src/core/iomgr/tcp_posix.c 6
```

results in
```
tag 50%/90%/95%/99% us
205 5.00/5.00/5.00/5.00

Important marks:
================
999999 @ src/core/iomgr/tcp_posix.c:3
205 { (src/core/iomgr/tcp_posix.c:1):        2.000 us
205 { (src/core/iomgr/tcp_posix.c:2):        1.000 us
205 } (src/core/iomgr/tcp_posix.c:4):       -1.000 us
205 } (src/core/iomgr/tcp_posix.c:6):       -3.000 us

999999 @ src/core/iomgr/tcp_posix.c:5
205 { (src/core/iomgr/tcp_posix.c:1):        4.000 us
205 } (src/core/iomgr/tcp_posix.c:6):       -1.000 us
```

### CODE REVIEW

**Comment by dgquintas** [comment]
Next step is to present broken down timings for the regular begin-end marks.

File: tools/profile_analyzer/profile_analyzer.py [hunk 1]
```diff
@@ -66,13 +90,27 @@ def entries():
...
+# Indexed by the mark's tag. Items in the value list correspond to the mark in
+# different stack situations.
+important_marks = collections.defaultdict(list)
+
 for entry in entries():
   thread = threads[entry.thread]
   if entry.type == '{':
     thread[entry.tag].append(entry)
-  elif entry.type == '}':
+  if entry.type == '}':
```
Comment (ctiller) [inline tools/profile_analyzer/profile_analyzer.py]: Keep the elif here perhaps?

File: tools/profile_analyzer/profile_analyzer.py [hunk 2]
```diff
@@ -80,8 +118,21 @@ def percentile(vals, pct):
...
 for tag in sorted(times.keys()):
   vals = times[tag]
-  print '%d %.2f/%.2f/%.2f/%.2f' % (tag, 
+  print '%d %.2f/%.2f/%.2f/%.2f' % (tag,
                                     percentile(vals, 50),
                                     percentile(vals, 90),
                                     percentile(vals, 95),
                                     percentile(vals, 99))
+
+print '\nImportant marks:'
```
Comment (ctiller) [inline tools/profile_analyzer/profile_analyzer.py]: maybe

print
print 'Important marks:'

for consistency elsewhere

**Comment by ctiller** [comment]
Bikeshedding comments included :)

This made my day.

**Comment by dgquintas** [comment]
Comments addressed. PTAL.

**Pull request merged**